### PR TITLE
Add RMF deep link navigation

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206596674293772/f

### Description
Adds the ability to perform navigation via RMF.

### Steps to test this PR

_Point to the RMF config linked in the task_
- [ ] Open the app.
- [ ] Verify that RMF is parsed correctly.

### CODE REVIEW

**Comment by joshliebe** [comment]
* **#4393** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4393?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joshliebe and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by cmonfortep — COMMENTED** [review]
Leaving some changes to address about payload (should be optional).

Can you merge rest of the stack onto this one so I can test E2E after applying the changes? Thanks!

File: navigation/navigation-api/src/main/java/com/duckduckgo/navigation/api/GlobalActivityStarter.kt [hunk 1]
```diff
@@ -148,7 +148,7 @@ private const val ACTIVITY_SERIALIZABLE_PARAMETERS_ARG = "ACTIVITY_SERIALIZABLE_
  * @ContributeToActivityStarter(ExampleActivityParams::class)
  * class ExampleActivity() : DuckDuckGoActivity() {
  *   fun onCreate(...) {
- *     val params = intent.getActivityParams(ExampleActivityParams::class)
+ *     val params = intent.getActivityParams(ExampleActivityParams::class.java)
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply navigation/navigation-api/src/main/java/com/duckduckgo/navigation/api/GlobalActivityStarter.kt:151]: The `.java` is required (That’s why I updated the documentation). For example: https://github.com/duckduckgo/Android/blob/3b3e9085ba797d437207fd697707d48933b17149/app/src/main/java/com/duckduckgo/app/browser/webview/WebViewActivity.kt#L57

File: browser-api/src/main/java/com/duckduckgo/app/accessibility/AccessibilityScreens.kt [hunk 2]
```diff
@@ -22,3 +22,8 @@ import com.duckduckgo.navigation.api.GlobalActivityStarter.ActivityParams
  * Use this model to launch the Accessibility screen
  */
 object AccessibilityScreenNoParams : ActivityParams
+
+/**
+ * Use this model to launch the Accessibility screen with a highlighted item
+ */
+data class AccessibilityScreenParams(val highlightedItem: String) : ActivityParams
```
Comment (aitorvs) [inline browser-api/src/main/java/com/duckduckgo/app/accessibility/AccessibilityScreens.kt:29]: I'd change to the following
```suggestion
sealed class AccessibilityScreens : ActivityParams {
    /**
     * Use this model to launch the Accessibility screen
     */
    data object Default : AccessibilityScreens() {
        private fun readResolve(): Any = Default
    }

    /**
     * Use this model to launch the Accessibility screen with a highlighted item
     */
    data class HighlightedItem(val highlightedItem: String) : AccessibilityScreens()
}
```

Comment (joshliebe) [reply browser-api/src/main/java/com/duckduckgo/app/accessibility/AccessibilityScreens.kt]: Updated in: https://github.com/duckduckgo/Android/pull/4393/commits/7abd2b467982141303e21fff1dfad4585a46538b

**Review by cmonfortep — APPROVED** [review]
LGTM thanks @joshliebe great work!

**Pull request merged**